test/In.ProjectEKA.HipServiceTest/Discovery/PatientControllerTest.cs

```diff
@@ -0,0 +1,477 @@
+
+            actualResponse = responsesSentToGateway[discoveryRequest.TransactionId];
+        }
+
+        private static void AndTheSentResponseShouldContainTheFoundPatient(GatewayDiscoveryRepresentation actualResponse,
+            PatientEnquiryRepresentation patientEnquiry)
+        {
+            actualResponse.Patient.ReferenceNumber.Should().Be(patientEnquiry.ReferenceNumber);
+            actualResponse.Patient.Display.Should().Be(patientEnquiry.Display);
+            actualResponse.Patient.CareContexts.Count().Should().Be(patientEnquiry.CareContexts.Count());
+            foreach (CareContextRepresentation careContext in patientEnquiry.CareContexts)
+            {
+                actualResponse.Patient.CareContexts.Should().ContainEquivalentOf(careContext);
+            }
+        }
+
+        private static void AndTheResponseShouldNotContainAnyPatientDetails(GatewayDiscoveryRepresentation actualResponse)
+        {
+            actualResponse.Patient.Should().BeNull();
+        }
+
+        private static void AndTheResponseShouldContainTheMatchFields(GatewayDiscoveryRepresentation actualResponse,
+            ICollection<string> matchedFields)
+        {
+            actualResponse.Patient.MatchedBy.Count().Should().Be(matchedFields.Count);
+            foreach (var matchedFieldName in matchedFields)
+            {
+                actualResponse.Patient.MatchedBy.Should().ContainEquivalentOf(matchedFieldName);
+            }
+        }
+
+        private static void AndTheResponseShouldNotContainAyMatchFields(GatewayDiscoveryRepresentation actualResponse)
+        {
+            actualResponse.Patient.Should().BeNull();
+        }
+
+        private static void AndTheResponseShouldContainTheTransactionId(GatewayDiscoveryRepresentation actualResponse,
+            DiscoveryRequest discoveryRequest)
+        {
+            actualResponse.TransactionId.Should().Be(discoveryRequest.TransactionId);
+        }
+
+        private static void AndTheResponseShouldContainTheErrorDetails(GatewayDiscoveryRepresentation actualResponse,
+            ErrorRepresentation errorRepresentation)
+        {
+            actualResponse.Error.Code.Should().Be(errorRepresentation.Error.Code);
+            actualResponse.Error.Message.Should().Be(errorRepresentation.Error.Message);
+        }
+
+        private static void AndTheResponseShouldNotContainAnyError(GatewayDiscoveryRepresentation actualResponse)
+        {
+            actualResponse.Error.Should().BeNull();
+        }
+
+        private static void AndTheResponseShouldContainTheExpectedStatus(GatewayDiscoveryRepresentation actualResponse, 
+            DiscoveryRequest discoveryRequest, HttpStatusCode expectedStatusCode, string expectedMessage)
+        {
+            
+            actualResponse.Resp.RequestId.Should().Be(discoveryRequest.RequestId);
+            // actualResponse.Resp.StatusCode.Should().Be(expectedStatusCode);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6377136)